File: Lecture6/VariableNameRandomGen.java
import java.util.Scanner;
import java.util.Random;

public class VariableNameRandomGen {
    public static void main(String[] args) {

        Scanner scan = new Scanner(System.in);
        Random rand = new Random();

        System.out.print("Length of the variable? ");
        int length = scan.nextInt();

        int num = rand.nextInt(64);
        if ( num < 26 )
            System.out.print(RandomCharacter.getRandomLowerCaseLetter());
        else if ( num < 52 )
            System.out.print(RandomCharacter.getRandomUpperCaseLetter());
        else if ( num < 62 );
        else if ( num == 62 ) System.out.print('_');
        else System.out.print('$');

        for ( int i = 0; i < length-1; i++ ){
            num = rand.nextInt(64);
            if ( num < 26 )
                System.out.print(RandomCharacter.getRandomLowerCaseLetter());
            else if ( num < 52 )
                System.out.print(RandomCharacter.getRandomUpperCaseLetter());
            else if ( num < 62 )
                System.out.print(RandomCharacter.getRandomDigitCharacter());
            else if ( num == 62 ) System.out.print('_');
            else System.out.print('$');
        }

        System.out.println();
    }
}

File: Lecture6/VariableNameRandomGen.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

char getRandomCharacter(char, char);
char getRandomLowerCaseLetter();
char getRandomUpperCaseLetter();
char getRandomDigitCharacter();
char getRandomCharacter_1();

int main(void) {
    srand(time(NULL));

    printf("Length of the variable? ");
    int length; scanf("%d", &length);

    int num = rand() % 63;
    if ( num < 26 )
        printf("%c", getRandomLowerCaseLetter());
    else if ( num < 52 )
        printf("%c", getRandomUpperCaseLetter());
    else if ( num < 62 );
    else printf("_");

    for ( int i = 0; i < length-1; i++ ){
        num = rand() % 63;
        if ( num < 26 )
            printf("%c", getRandomLowerCaseLetter());
        else if ( num < 52 )
            printf("%c", getRandomUpperCaseLetter());
        else if ( num < 62 )
            printf("%c", getRandomDigitCharacter());
        else printf("_");
    }

    printf("\n");

    return 0;
}

char getRandomCharacter(char ch1, char ch2) {
    return ch1 + rand() % (ch2 - ch1 + 1);
}

char getRandomLowerCaseLetter() {
    return getRandomCharacter('a', 'z');
}

char getRandomUpperCaseLetter() {
    return getRandomCharacter('A', 'Z');
}

char getRandomDigitCharacter() {
    return getRandomCharacter('0', '9');
}

char getRandomCharacter_1() {
    return getRandomCharacter(0, 127);
}
